binderhub/build.py

```diff
@@ -382,8 +383,24 @@ def submit(self):
 
         env = []
         if self.git_credentials:
+            secret_content = base64.b64encode(
+                self.git_credentials.encode("utf-8")
+            ).decode("utf-8")
+            data = {"credentials": secret_content}
+
+            secret = client.V1Secret()
```

Comment:
When creating a k8s Secret:

If you have data that is a string, you can often just store it directly as a string using `stringData` instead of `data` in the k8s Secret resource's `spec`. `data` require you to provide data that has been base64 encoded, while `stringData` doesn't.

If you are parsing the data later, the k8s Secret resource will only have a `data` field, as it will have merged all data into the `data` section.

Using `stringData` here is probably a bit simpler, maybe go for that?

https://kubernetes.io/docs/concepts/configuration/secret/#restriction-names-data

Reply:
Indeed, excellent suggestion, I will update the patch.